lib/lz4frame.c

```diff
@@ -394,9 +395,41 @@ size_t LZ4F_compressFrame(void* dstBuffer, size_t dstCapacity,
                     const void* srcBuffer, size_t srcSize,
                     const LZ4F_preferences_t* preferencesPtr)
 {
-    return LZ4F_compressFrame_usingCDict(dstBuffer, dstCapacity,
-                                         srcBuffer, srcSize,
-                                         NULL, preferencesPtr);
+    size_t result;
+#if (LZ4_HEAPMODE)
+    LZ4F_cctx_t *cctxPtr;
+    LZ4F_createCompressionContext(&cctxPtr, LZ4F_VERSION);
```

Comment:
check if context creation is actually successful

Reply:
This comment is actually pretty important.
You really have to check that `cctxPtr != NULL`,
otherwise, the next instruction, which is `LZ4F_compressFrame_usingCDict()`, will segfault